app/master/cluster_master.py

```diff
@@ -279,7 +282,7 @@ def handle_result_reported_from_slave(self, slave_url, build_id, subjob_id, payl
         build = self._all_builds_by_id[int(build_id)]
         slave = self._all_slaves_by_url[slave_url]
         # If the build has been canceled, don't work on the next subjob.
-        if not build.is_finished:  # WIP(joey): This check should be internal to the Build object.
+        if not build.is_finished or build.has_error:  # WIP(joey): This check should be internal to the Build object.
```

Comment:
This seems like it would make builds that have failed due to an error keep executing, which I think is undesirable. I kind of suspect they added this to better deal with some other weird behavior around build cancellation (maybe builds getting stuck that are canceled before any subjobs start). I think we should leave this out, especially because we are also going to be fixing the issues with build cancellation.